### DISCUSSION TITLE
Between operation in query.where(), with table types float end with TypeError: Boolean value of this clause is not defined

### DISCUSSION THREAD

**Discussion by b0r1ngx** [sqlalchemy/sqlalchemy#10641] [Usage Questions] [answered]
Screenshot is: ![image](https://github.com/sqlalchemy/sqlalchemy/assets/55315908/168c5ad4-dd04-4691-9aa5-f366b701ad8c)

Im using sqlite driver

You can see in console that here is error.
Operation that happen is on center (left side)
And table Profile is on right side

Please check this out, and help me if you know how to

Maybe something bad in 2.0? Or its just doesn't work before?

**Comment by zzzeek** [comment, marked as answer]
it would be more helpful if you use pasted code samples in general, however I can see in your screenshot you are using the Python `and` operator inside of a SQLAlchemy SQL expression, which is going to evaluate as boolean, which then raises here.  The raise is explicitly so you can catch this mistake.  you want to use [and_()](https://docs.sqlalchemy.org/en/20/core/sqlelement.html#sqlalchemy.sql.expression.and_).

**Reply by b0r1ngx** [reply]
Oh, I got you, its maybe error with this operator..., I try to!

I try to google with this error and I found something like this but: https://stackoverflow.com/questions/42681231/sqlalchemy-unexpected-results-when-using-and-and-or 

Accepted answer is trying to give me a point on this operators, but I just don't got this and think that is not my case...

**Reply by b0r1ngx** [reply]
Yes dude, this is it, I fix it, and now it works pretty well, hope I don't get more of this... 

So stuck with this new SQLAlchemy 2.0 fixes... (even if this is also come from python & SQLAlchemy 1.x)

Screenshot for someone, who doesn't mind what we talking about, and want to see fixes in code/screen: 
![image](https://github.com/sqlalchemy/sqlalchemy/assets/55315908/2ebdfe29-74d3-46d5-9892-69fc50536ed6)

**Reply by b0r1ngx** [reply]
And I got you, with pasted parts of code. 

I've just thinking that this will be enough, but in some cases, also if you want try to implement this on your machine, it would be more helpful.

Great Thanks!:) Good Luck!

**Reply by CaselIT** [reply]
note that this is not a change 1.x -> 2.0.
the `and` operation has always evaluated to True / False in python.
What has changed is that 2.0 may raise an error here, while 1.x rendered the wrong query (either `where true` or `where false`)
